packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2822,26 +2822,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // The call to fetch the snapshot is very expensive and not always needed.
         // It should only be done by the summarizerNode, if required.
         const snapshotTreeFetcher = async () => {
+            // When fetching from storage we will always get the latest version and do not use the ackHandle.
             const fetchResult = await this.fetchSnapshotFromStorage(
-                ackHandle,
-                summaryLogger,
+             null,
+             summaryLogger,
+             {
+                 eventName: "RefreshLatestSummaryGetSnapshot",
+                 ackHandle,
+                 summaryRefSeq,
+                 fetchLatest: true,
+             });
+             return fetchResult.snapshotTree;
+         };
+         const result = await this.summarizerNode.refreshLatestSummary(
+             proposalHandle,
+             summaryRefSeq,
+             snapshotTreeFetcher,
+             readAndParseBlob,
+             summaryLogger,
+         );
+
+         // In case we had to retrieve the latest snapshot and it is different than summaryRefSeq,
+         // wait for the delta manager to catch up.
+         if (result.latestSummaryUpdated === true && result.wasSummaryTracked === false) {
+            assert(result.snapshot !== undefined, "should have tree result");
+            const latestSnapshotRefSeq = await seqFromTree(result.snapshot, readAndParseBlob);
+            if (latestSnapshotRefSeq !== summaryRefSeq) {
+                summaryLogger.sendTelemetryEvent(
                 {
-                    eventName: "RefreshLatestSummaryGetSnapshot",
-                    ackHandle,
-                    summaryRefSeq,
-                    fetchLatest: false,
+                    eventName: "LatestSummaryRetrieved",
+                    lastSequenceNumber: latestSnapshotRefSeq,
+                    targetSequenceNumber: summaryRefSeq,
                 });
-            return fetchResult.snapshotTree;
-        };
-        const result = await this.summarizerNode.refreshLatestSummary(
-            proposalHandle,
-            summaryRefSeq,
-            snapshotTreeFetcher,
-            readAndParseBlob,
-            summaryLogger,
-        );
+                if (latestSnapshotRefSeq > this.deltaManager.lastSequenceNumber) {
+                    // We need to catch up to the latest summary's reference sequence number before pausing.
+                    await PerformanceEvent.timedExecAsync(
```

Comment:
While waiting for catch up, more acks can come in and they will queue the same thing. That should be prevented.

Reply:
Maybe it's related or not, but I had somewhat similar question handleSummaryAcks() loading all summaries (in case summarizer loaded from cached summary). I think it should not matter - mostly because Summarizer.runCore() will wait for summarizer to be connected (and thus fully caught up) before handleSummaryAcks is even called, so we always start with last summary and proceed forward.

There could be other summarizers forcing handleSummaryAcks() to refresh knowledge, but that seems like it should be so rare (at least at the moment) that we should not care, no?

Reply:
If we go with catching up first and then calling refresh on summarizer node, then we need to handle this function being called multiple times.

Reply:
I do not think it's the case. Take a look at the only caller - handleSummaryAcks(). It will execute only one refresh at a time) plus we use lock to ensure summaries are not running during this time.).

Assuming that's correct, I have one related question (and it's related to [earlier question](https://github.com/microsoft/FluidFramework/pull/12313#issuecomment-1270860197) I asked): Why do we need to have handleSummaryAcks() in the form it exists today? It feels like we combine different processes here and it would be great to separate them (maybe in separate PR):
1. Pending summary tracking - I feel like it should do summarizer code (the one that submits summary) only. It should realize it failed to summarize even if there is no Nack, but some other summary Ack showed up with a later seq number.
2. Learning about ack for the summary we loaded from / latest summary - I feel like it's mostly noop, given that we start listening for ops after we fully catch up. In case of race condition (summary ack and join op - possible) - follow next item.
3. Any "unknown" summary ack shows up - fail-fast by closing summarizer and restarting.

Purpose - reduce number of states we support / track / allow and move corner case handling to simple client reboot.

Reply:
I agree that this function won't be called multiple times today because of the way handleSummaryAcks() works. However, this method should be self-sufficient and not rely on callers during the right thing. If we do rely on that, this function should mention that (in comments) and assert that it is not called again while in progress.

Reply:
#1 is rare. I don't think it can even happen right? We pause incoming ops before we start to summarize.
#2 is already a no-op. The summarizer node refresh summary function ignores it.
#3 is problematic. This is common scenario with snapshot caching. Summarizer client can load from an older snapshot and will get a bunch of summary ops and summary acks while catching up.

Reply:
#1 - we can get a summary ack before our summarization even starts. if we get a summary ack with a handle that we do not expect, and it has summaryRefSeq < our current reference seq, it will be a no-op. If it has summaryRefSeq > our current reference seq, we will get the latest one and catch up if required. 
#2 agree with Navin, it is already a no-op. 
#3  - why would we do that ? It should be fine to start from the latest known summary - wouldn't it ? 

I suppose the only real use for the handleSummaryAcks is for the scenario where we load the snapshot from cache and the summaryack op arrives to let us know there are new snapshots available. From that moment on and with this current change , we will get the latest snapshot (no matter how many behind we are) and new Summary Acks will be a no-op as our reference number will be greater than their reference number.

Reply:
@NicholasCouri - talked to Navin. I believe my earlier point about ensuring we have test that demonstrates the problem (i.e. fails without the rest of your changes) is a must here. I believe that earlier sequence I put in this discussion (1-3) is correct (due to await this.runCoordinatorCreateFn() in Summarizer.runCore()) that waits for "connected" event before handleSummaryAcks(). But maybe there is a bug, maybe my understanding is wrong - we need to clearly see that (describe it in plain English, as well as test coverage) that demonstrates the problem first.

If I'm right - we download only one summary by ID today. It's a problem worth fixing (this PR), but I think there were reports where we were downloading many summaries by ID, especially when loading summarizer from caches snapshot in the past. I do not see how that later case is possible - code should still load just one summary and enabling back caching for summarizer is actually good thing to return if we can prove it makes system better (in average case).

I think we need number of test cases (assuming we do not have them)

case 1:
- creates a bunch of summaries
- load summarizer from latest summary (with proper summary op/ack)
- validate client does not fetch any summary - it can match summary ack with its own snapshot and realize it has all the data it needs.

case 2:
- Same as above, but there is no summary op / ack for latest summary (or we hide it somehow from summarizer).
- validate. Not sure what it means in practice - I feel like client should assume there are no more summaries move on (i.e. not wait 10 min for ack, ignoring the race that condition where summary op / ack may show up later). If it fails to produce a summary later - it can recover (hopefully by reboot).

case 3:
- creates a bunch of summaries
- load summarizer from old summary (Loader supports loading from version).
- Validate client can fetch only latest summary (and only one!) to move on.

Reply:
I've added a couple of test cases (1,2,3) and will add more afterwards